Homework4/task1/task1.c

```diff
@@ -0,0 +1,126 @@
+﻿#include <stdio.h>
+#include <locale.h>
+#include <stdbool.h>
+#include <malloc.h>
+#include <math.h>
+#include <stdlib.h>
+#define BINARY_SIZE sizeof(int) * 8
+
+int verificationIntScanf(void) {
+    int readValues = 0;
+    while (true) {
+        int correctlyReadValues = scanf("%d", &readValues);
+        while (getchar() != '\n') {
+        }
+        if (correctlyReadValues == 1) {
+            break;
+        }
+        printf("Ошибка ввода. Введите целочисленное значение!\n");
+    }
+    return readValues;
+}
+
+void printBinary(const short* binaryNumber, const unsigned int size) {
+    for (int i = 0; i < size; ++i) {
+        printf("%d", binaryNumber[i]);
+        if ((i + 1) % 8 == 0 ) {
+            printf(" ");
+        }
+    }
+    printf("\n");
+}
+
+void binaryRepresentation(int number, short* representation, const unsigned int size) {
+    for (int i = size - 1; i >= 0; --i) {
+        representation[i] = (short)(number & 1);
+        number = number >> 1;
+    }
+}
+
+void binarySum(const short* firstBinary, const short* secondBinary, short* sum, const unsigned int size) {
+    bool overflow = false;
+    for (int i = size - 1; i >= 0; --i) {
+        sum[i] = (short) (firstBinary[i] ^ secondBinary[i] ^ overflow);
+        overflow = (firstBinary[i] + secondBinary[i] + overflow) > 1 ? true : false;
```

Comment:
```suggestion
        overflow = (firstBinary[i] + secondBinary[i] + overflow) > 1;
```
Это и так булевое значение ведь